LocalParks.API/Controllers/ApiSupervisorsController.cs

```diff
@@ -0,0 +1,70 @@
+﻿using LocalParks.Core.Contracts;
+using LocalParks.Core.Models;
+using Microsoft.AspNetCore.Authentication.JwtBearer;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Extensions.Logging;
+using System;
+using System.Threading.Tasks;
+
+namespace LocalParks.API.Controllers
+{
+    [ApiController]
+    [Authorize(AuthenticationSchemes = JwtBearerDefaults.AuthenticationScheme)]
+    public class SupervisorsController : ControllerBase
+    {
+        private readonly ILogger<SupervisorsController> _logger;
+        private readonly ISupervisorsService _service;
+
+        public SupervisorsController(ILogger<SupervisorsController> logger,
+            ISupervisorsService service)
+        {
+            _logger = logger;
+            _service = service;
+        }
+
+        [HttpGet("api/[controller]")]
+        public async Task<ActionResult<SupervisorModel[]>> GetAllSupervisors()
+        {
+            _logger.LogInformation("API GET request: All Supervisors");
+
+            try
+            {
+                var results = await _service.GetAllSupervisorModelsAsync();
+
+                if (results == null) return NoContent();
+
+                return Ok(results);
+            }
+            catch (Exception ex)
+            {
+                _logger.LogError($"Error occured in getting all Supervisors: {ex.Message}");
+
+                return StatusCode(StatusCodes.Status500InternalServerError, $"Database Failure");
+            }
+        }
+
+        [HttpGet("api/[controller]/{parkId:int}")]
+        [HttpGet("api/parks/{parkId:int}/[controller]")]
```

Comment:
two httpget attributes?

Reply:
Thought was to have the action accessible from two endpoints for option of how to query it.
Have removed the latter now as it is not used anyway